### ISSUE TITLE
VGG difference from paper

### ISSUE THREAD

**Issue by msalvaris** [keras-team/keras#9335]
Hi, just wanted to ask whether there was a reason the implmentation in Keras of VGG differed from the one in the paper [Very Deep Convolutional Networks for Large-Scale Image Recognition](https://arxiv.org/abs/1409.1556)? It looks like the regularization and dropout are missing. In Pytorch they have the dropout in the FC layers https://github.com/pytorch/vision/blob/master/torchvision/models/vgg.py. The paper is a little unclear since in Table 1 it looks like the FC layers connect to each other but if you read section 3.1 Training it says "The training was regularised by weight decay (the L2 penalty multiplier set to 5·10−4) and dropout regularisation for the first two fully-connected layers (dropout ratio set to 0.5)"

**Comment by ndrplz** [comment]
Bumped into the same issue.

Apparently it is because keras.applications is designed for inference in mind (#9719), but I find this choice a bit confusing too.
